src/pages/LenderProfile/LenderProfile.vue

```diff
@@ -93,21 +95,27 @@ export default {
 			lenderInfo: {},
 			publicId: '',
 			lenderIsPublic: false,
-			enableBadgeContent: false,
-			badgeImage: null,
-			badgeCategory: '',
-			badgeTarget: 0,
+			badgeData: null,
 		};
 	},
 	apollo: {
 		preFetch(config, client, { route }) {
 			const currentRoute = route.value ?? route ?? {};
 			const publicId = currentRoute.params?.publicId ?? '';
-			const { isBadgeKeyValid, badgeQuery, badgeKey } = useBadgeContentfulData(client, currentRoute);
+			const badgeKey = currentRoute.query?.utm_campaign;
+			const { isBadgeKeyValid } = useBadgeData();
 
 			return Promise.all([
 				client.query({ query: lenderPublicProfileQuery, variables: { publicId } }),
-				isBadgeKeyValid ? client.query({ query: badgeQuery, variables: { badgeKey } }) : null,
+				isBadgeKeyValid(badgeKey) ? [
```

Comment:
It looks like this still runs the queries separately. Also, does this work? I think you would still need to spread the array created here. Is it possible to combine these queries to reduce the # of queries called?